.github/workflows/ci.yml

```diff
@@ -76,3 +76,31 @@ jobs:
         run: npm ci --no-audit --no-progress --no-optional
       - name: Build
         run: npm run build
+
+        # Only build and push docker image if this action is triggered from the main branch.
+
+      - name: Build Docker image
+        if: ${{ github.ref == 'refs/heads/main' }}
+        run: docker build . --file Dockerfile --tag $IMAGE_NAME --label "runnumber=${GITHUB_RUN_ID}"
+
+      - name: Log in to Github Package registry
+        if: ${{ github.ref == 'refs/heads/main' }}
+        run: echo "${{ secrets.GITHUB_TOKEN}}" | docker login ghcr.io -u $ --password-stdin
+
+      - name: Push image to registry
+        if: ${{ github.ref == 'refs/heads/main' }}
+        run: |
+          IMAGE_ID=ghcr.io/${{ github.repository_owner }}/$IMAGE_NAME
+          
+          # Change all uppercase to lowercase
+          IMAGE_ID=$(echo $IMAGE_ID | tr '[A-Z]' '[a-z]')
+          # Strip git ref prefix from version
+          VERSION=$(echo "${{ github.ref }}" | sed -e 's,.*/\(.*\),\1,')
+          # Strip "v" prefix from tag name
+          [[ "${{ github.ref }}" == "refs/tags/"* ]] && VERSION=$(echo $VERSION | sed -e 's/^v//')
+          # Use Docker `latest` tag convention
+          [ "$VERSION" == "main" ] && VERSION=latest
+          echo IMAGE_ID=$IMAGE_ID
+          echo VERSION=$VERSION
+          docker tag $IMAGE_NAME $IMAGE_ID:$VERSION
+          docker push $IMAGE_ID:$VERSION
```

Comment:
These changes shouldn't be part of your PR so they must have slipped in while you were working?

Reply:
I am not sure, these two GitHub workflow files changed after I Merged the main into this branch.

Reply:
Let me know if i should resolve this conversation or is there anything I have to fix?